docs/newtypes/index.md

```diff
@@ -0,0 +1,310 @@
+
+The easiest way to do this is by defining our operators as extension methods inside the new type companion object. These extension methods will automatically be available when we are working with instances of the new type.
+
+```scala mdoc:reset-object
+import zio.prelude.Newtype
+
+object SequenceNumber extends Newtype[Int] {
+
+  implicit class SequenceNumberSyntax(private val self: SequenceNumber) extends AnyVal {
+    def next: SequenceNumber =
+      SequenceNumber.wrap(SequenceNumber.unwrap(self) + 1)
+  }
+}
+type SequenceNumber = SequenceNumber.Type
+
+val sequenceNumber: SequenceNumber =
+  SequenceNumber(1)
+
+val nextSequenceNumber: SequenceNumber =
+  sequenceNumber.next
+```
+
+Notice that the `next` operator was available on `SequenceNumber` without having to do any imports because we defined it as an extension method within the `SequenceNumber` object.
+
+The new type companion object is also a convenient place for other operators that we would normally define on the companion object of a type. For example, we could define a `continuous` operator that returns whether a collection of `SequenceNumber` values are continuous.
+
+```scala mdoc:nest
+import zio.Chunk
+
+object SequenceNumber extends Newtype[Int] {
+  def continuous(sequenceNumbers: Iterable[SequenceNumber]): Boolean =
+    SequenceNumber.unwrapAll(sequenceNumbers).foldLeft[(Option[Int], Boolean)]((None, true)) {
+      case ((None, continuous), current) => (Some(current), continuous)
+      case ((Some(previous), continuous), current) => (Some(current), continuous && current == previous + 1)
+    }._2
+}
+type SequenceNumber = SequenceNumber.Type
+
+val sequenceNumbers: Chunk[SequenceNumber] =
+  Chunk(SequenceNumber(1), SequenceNumber(2), SequenceNumber(3))
+
+val continuous: Boolean =
+  SequenceNumber.continuous(sequenceNumbers)
+```
+
+Here `continuous` is an operator on a collection of sequence numbers so it didn't make sense to define it as an extension method on an individual sequence number. Defining it in the new type companion object makes it feel like an operator on the companion object of any ordinary data type.
+
+## Defining Instances Of Functional Abstractions For New Types
+
+We can also define instances of functional abstractions for new types such as `Equal`, `Ord`, and `Associative`.
+
+The new type companion object again provides a convenient place to define these instances and ensures that they will automatically be available so that we don't have to import them.
+
+Let's see how we can do this to define an `Equal` instance for `AccountNumber`:
+
+```scala mdoc:reset-object
+import zio.prelude._
+
+object AccountNumber extends Newtype[Int] {
+  implicit val AccountNumberEqual: Equal[AccountNumber] =
+    Equal.default
+}
+type AccountNumber = AccountNumber.Type
+
+AccountNumber(1) === AccountNumber(1) // true
+AccountNumber(1) === AccountNumber(2) // false
+// AccountNumber(1) === 1             // does not compile
+```
```

Comment:
not sure if I am correct: you could add a `AccountNumber(1) == 1 // true, compiles, but misleading` and discuss?

Reply:
I think I discuss that right below?